k4EDM4hep2LcioConv/src/k4Lcio2EDM4hepConv.cpp

```diff
@@ -288,15 +260,65 @@ namespace LCIO2EDM4hepConv {
     return dest;
   }
 
+  std::vector<CollNamePair> convertLCIntVec(const std::string &name, EVENT::LCCollection *LCCollection) {
```

Comment:
```suggestion
  template<typename LCVecType>
  std::vector<CollNamePair> convertLCVec(const std::string &name, EVENT::LCCollection *LCCollection) {
```
This could be a template that handles `LCIntVec` and `LCFloatVec` with only one implementation. Internally, we would then need to do something along the lines of
```cpp
auto dest = std::make_unique<podio::UserDataCollection<LCVecType::value_type>>>();
// ... unchanged until
  const auto* rval = static_cast<LCVecType*>(LCCollection->getElementAt(i));
// ... unchanged until return
```